pom.xml

```diff
@@ -15,7 +15,7 @@
 
     <groupId>com.lazerycode.jmeter</groupId>
     <artifactId>jmeter-maven-plugin</artifactId>
-    <version>DEV-SNAPSHOT</version>
+    <version>3.3.0</version>
```

Comment:
This should be DEV-SNAPSHOT

Reply:
Yes, I dit not think about it :-/ 
Your call cover most of my use cases. The only thing is that in original code the selectedConfiguration is don't use because mojo always use executionId to find config. But in my code your call will not work:-/ so backward compability is not preserved. I will close this PR.

Reply:
I think that code below is not possible now (selectedConfiguration  in gui is not used) in current plugin code (two the same execution ids are forbidden - so I can't use <id>configuration2</id> in gui goal) so it will not work: mvn pre-integration-test :


          <!-- Generate JMeter configuration -->
          <execution>
            <id>configuration2</id>
            <goals>
              <goal>configure</goal>
            </goals>
            <configuration>
              <jmeterExtensions>
                <artifact>kg.apc:jmeter-plugins-standard:1.4.0</artifact>
              </jmeterExtensions>
              <testPlanLibraries>
                <artifact>org.example:someArtifact:1.1-SNAPSHOT</artifact>
              </testPlanLibraries>
              <testFilesIncluded>some_jmeter_test.jmx</testFilesIncluded>
            </configuration>
          </execution>
          <!-- Run JMeter tests -->
          <execution>
            <id>jmeter-tests</id>
            <goals>
              <goal>jmeter</goal>
            </goals>
              <configuration>
                  <selectedConfiguration>configuration2</selectedConfiguration>
              </configuration>
          </execution>
          <!-- Fail build on errors in test -->
          <execution>
            <id>jmeter-check-results</id>
            <goals>
              <goal>results</goal>
            </goals>
          </execution>
		  <!-- run gui -->
            <execution>
                <id>gui</id>
                <phase>pre-integration-test</phase>
                <goals>
                    <goal>gui</goal>
                </goals>
                <configuration>
                    <selectedConfiguration>configuration2</selectedConfiguration>
                </configuration>
            </execution>

But it may be not offen used use case ...and another PR I think to make selectedConfiguration works in gui.

Reply:
So I'm not 100% I'm understanding what you are saying?  Maven will allow you to specify the execution ID using the syntax above so:

```mvn jmeter:gui@configuration```

will start the GUI using an execution id of `configuration`

If you wanted to use configuration2 you would instead use:

```mvn jmeter:gui@configuration2```

You don't need to modify the POM

Reply:
I have thought also about pom.xml like example I pasted above where everything is in pom.xml even gui goal with phase redefinition and when I run mvn as:
`mvn pre-integration-test `  (look example above)
I  got:

> 
> No results for path: $[0]
> [ERROR] Have you added the configure goal to your POM?
> [ERROR]     <execution>
> [ERROR]         <id>configuration</id>
> [ERROR]         <goals>
> [ERROR]             <goal>configure</goal>
> [ERROR]         </goals>
> [ERROR]     </execution>

In above pasted example gui can't find configuration because it take it from execution id - and I can't change it in pom to configuration2 as defined in configuration goal because maven don't allow two equal executionids and `selectedConfiguration>configuration2</selectedConfiguration>`  in gui goal is not working.

Your call `mvn jmeter:gui@configuration2` works if I earlier run `mvn jmeter:configure@configure2`  

In my case I would like to get extensions in jmeter\lib\ext and some artifacts to jmeter\lib which are defined in jmeter:configure in pom - to achieve i I can now use `mvn jmeter:configure@configuration2 jmeter:gui@configuration2`  thanks yo your first hint with redefining executionid from cli.